Ghost Hotel Update

Kush1Push1 opened the pull request:
## About The Pull Request

Added initial code to add variety of rooms in ghost hotel (a blue ball at Ghost Cafe).

Also, removed annoying bell sound inside rooms.

<details>

https://user-images.githubusercontent.com/78963858/192285182-526492b2-62b6-4313-a1d0-49d8aa1742a4.mp4

</details>

## How This Contributes To The Skyrat Roleplay Experience

A better roleplay experience in Ghost Cafe.

## Changelog

:cl: SishTis
add: Added rooms variety to ghost cafe's hotel.
add: Added fancy cobblestone pool floor. You can reskin pool floor tile in hand to get it.
del: Silenced bell ring in ghost cafe's hotel rooms.
/:cl:

GoldenAlpharex commented on code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:
```diff
@@ -7,6 +7,10 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 	w_class = WEIGHT_CLASS_SMALL
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/list/static/hotel_maps = list("Generic", "Apartment")
```
Add documentation to these two variables please.

Kush1Push1 replied:
Good enough? (In recent commit)

GoldenAlpharex commented on code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:
```diff
@@ -94,6 +101,11 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/chosen_room
+	if(istype(src, /obj/item/hilbertshotel/ghostdojo)) //to don't add another one var
+		chosen_room = tgui_input_list(user, "Choose desired room:","Time to choose", hotel_maps)
```
You need to handle the client not clicking any of the available options and just clicking X instead (which will return null in `chosen_room`).

Kush1Push1 replied:
Done

GoldenAlpharex commented on code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:
```diff
@@ -138,12 +150,18 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 	return FALSE
 
-/obj/item/hilbertshotel/proc/sendToNewRoom(roomNumber, mob/user)
+/obj/item/hilbertshotel/proc/sendToNewRoom(roomNumber, mob/user, chosen_room) //SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
```
You need to mention what that change you did is, so it's not a pain to figure out when it gets changed from upstream.

Kush1Push1 replied:
Done

GoldenAlpharex reviewed (changes requested):
I don't think that hidden room in the bottom-right corner should exist.

Furthermore, the walls you used aren't indestructible walls, which means anyone with a welder could just break into space with it. Which is very much not how it's intended.

The sink looks really awkward placed in the kitchen like that, and I also don't really know if it's really necessary in the first place.

The bathroom feels a bit weird, with the washing machine in there. I think it could use a bit more work, I'm not even sure that there's a sink and a mirror in there. You can take a bit more space for it, by removing the hidden room to its right.

The pool in the living room is a bit of a stretch, and I don't think it should be in there, it's too much.

Kush1Push1 commented:
<details>

![dreamseeker_HFUgBqwnAX](https://user-images.githubusercontent.com/78963858/192326960-d920637d-0fc3-4bf0-885a-8dcfd625ee28.png)

</details>

Kush1Push1 commented:
> I don't think that hidden room in the bottom-right corner should exist.

Done.

> Furthermore, the walls you used aren't indestructible walls, which means anyone with a welder could just break into space with it. Which is very much not how it's intended.

Replaced.

> The sink looks really awkward placed in the kitchen like that, and I also don't really know if it's really necessary in the first place.

Moved to other place.

> The bathroom feels a bit weird, with the washing machine in there. I think it could use a bit more work, I'm not even sure that there's a sink and a mirror in there. You can take a bit more space for it, by removing the hidden room to its right.

It's common for me, but alright. A sink and mirror were there.

> The pool in the living room is a bit of a stretch, and I don't think it should be in there, it's too much.

Moved.

Kush1Push1 commented:
@GoldenAlpharex ?

GoldenAlpharex commented on code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:
```diff
@@ -94,6 +103,11 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/chosen_room = "Nothing"
+	if(istype(src, /obj/item/hilbertshotel/ghostdojo)) //to don't add another one var
+		chosen_room = tgui_input_list(user, "Choose desired room:","Time to choose", hotel_maps)
```
```suggestion
		chosen_room = tgui_input_list(user, "Choose desired room:", "Time to choose", hotel_maps)
```

GoldenAlpharex commented on modular_skyrat/master_files/code/game/turfs/closed/_closed.dm:
```diff
@@ -0,0 +1,7 @@
+/turf/closed/indestructible/steel
```
I'd rename this to this, for less confusion (yes you'll have to update it in your map too, sorry!
```suggestion
/turf/closed/indestructible/iron
```

GoldenAlpharex reviewed (changes requested):
The wooden decals on the carpet look a bit off, because they're not uniform on any side and contrast heavily with the carpet's very defined edges.

I'm still entirely sold on the pool in the bathroom, unless you're telling me it's supposed to be a bath. In which case, it's a bit big for a bath, but I can see the reasoning. I'd just shrink it down a bit to make it more obvious.

Kush1Push1 commented:
> The wooden decals on the carpet look a bit off, because they're not uniform on any side and contrast heavily with the carpet's very defined edges.
> 
> I'm still entirely sold on the pool in the bathroom, unless you're telling me it's supposed to be a bath. In which case, it's a bit big for a bath, but I can see the reasoning. I'd just shrink it down a bit to make it more obvious.

I can't use "iron" at the end, it will override other code.

<details>

![Code_PH9P8LLimq](https://user-images.githubusercontent.com/78963858/194699917-f685d5ec-4f3f-464d-91c5-563ed8d33138.png)

![dreamseeker_cUYCvQSgo0](https://user-images.githubusercontent.com/78963858/194699918-1e2d6f98-254d-4a6a-8ea7-520a229be842.png)

</details>

I don't think I have enough will to work on it any further. I moved the bath and filled unused space with placeholder.

Wildwestmars commented:
My friend and I just got done playtesting the new apartments and...
The bathroom floods the entire complex as soon you open the door. There is also a floating painting frame in the living room.
Besides that, it was quite fun (and cozy)!

Kush1Push1 commented:
1) Fixed the map. The pool doesn't spill its water now and the paint was moved back on the wall (there was an issue with its direction).

2) Addead fancy subtype of pool, with look like cobblestone and make it craftable via reskin of pool floor tile.

Kush1Push1 commented:
<details>

The pool
![dreamseeker_juE5EomunT](https://user-images.githubusercontent.com/78963858/195836468-ebd1c622-f992-40a6-be48-bdbc4004d0ff.png)

The paint
![dreamseeker_Lf9nigeuTY](https://user-images.githubusercontent.com/78963858/195836494-1d475f31-8f9b-4e39-b8af-7776a3602cb0.png)

</details>

theselfish commented:
The bathroom needs a drain, the water spills and floods the entire room otherwise. Look at how the Cafe's pool does it for an example.

GoldenAlpharex reviewed (changes requested):
Apparently the bathroom needs a drain. Aside from that, it's all good with me, I'd say.

GoldenAlpharex commented:
I tried it myself and apparently that wasn't an issue, the only thing I've noticed is the fishes dying, but that's about it. It's good to send.
